packages/cli/src/command_middleware.ts

```diff
@@ -64,4 +66,19 @@ export class CommandMiddleware {
       throw new InvalidCredentialError(errMsg, { cause: err });
     }
   };
+
+  /**
+   * The AWS CDK respects older CLI v1 variable names that are no longer supported in the
+   * latest AWS SDK. Developers that use the older variables and interoperate between Amplify
+   * and CDK will experience region mismatch failures when using Amplify tools. Variable names
+   * known to cause such failures are normalized here for a better developer experience.
+   */
+  private normalizeEnvironmentVariables(): void {
```

Comment:
I think I'm okay with silently remapping this. The alternative would be to fail if AWS_DEFAULT_PROFILE or AWS_DEFAULT_REGION are set with a message saying to use AWS_PROFILE or AWS_REGION instead.

Regardless of that decision, I think we should error if both the old and new style of env var is set because that case is ambiguous which one the customer intends to use.

Reply:
Throwing was considered [see here](https://github.com/aws-amplify/amplify-backend/issues/965#issuecomment-2023927853) but the preference is to remap since there are some popular tools that set the older variants in sub shells.

I can get behind throwing when both are set.

Reply:
perhaps we should emit warning if legacy env vars are used.

Reply:
I decided to go with @sobolk suggestion and simply warn the customer when legacy env variables are detected. Hard failing when both are set is not necessary if we print what was mapped/preferred in the warning.